pyquil/paulis.py

```diff
@@ -899,6 +917,8 @@ def exponential_map(term: PauliTerm) -> Callable[[float], Program]:
     :param term: A pauli term to exponentiate
     :returns: A function that takes an angle parameter and returns a program.
     """
+    assert isinstance(term.coefficient, (float, complex))
```

Comment:
Is a `float` possible here? The type annotation on `PauliTerm.coefficient` above says it's `Union[Expression, complex]` and the `__init__` normalizes any `Number` to `complex`.

Reply:
I thought so too, but it happens in the unit tests. Unclear to me how, but I think that's a problem for another time